### PR TITLE
[BLZT-72] - @PreDestroy was added

### PR DESCRIPTION
**Pull request by proyalko**

### CODE REVIEW

File: core/src/test/java/com/bobocode/bring/core/predestroy/PredestroyTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      BringApplicationContext bringApplicationContext = BringApplication.run(TEST_DATA_PACKAGE + ".negative");
+      bringApplicationContext.close();
+    };
+
+    // then
+    PreDestroyException preDestroyException = assertThrows(PreDestroyException.class, executable);
+    assertThat(preDestroyException.getMessage()).isEqualTo(expectedMessage);
+
+
+    // given
```
Comment (PolukovY) [inline core/src/test/java/com/bobocode/bring/core/predestroy/PredestroyTest.java:50]: I think this part is just a copy of the previous test. Should be removed positive case.

File: core/src/main/java/com/bobocode/bring/core/exception/PreDestroyException.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.bobocode.bring.core.exception;
+
+public class PreDestroyException extends RuntimeException {
+  public static String PRE_DESTROY_EXCEPTION_MESSAGE ="@PreDestroy should be added to method without parameters";
```
Comment (PolukovY) [inline core/src/main/java/com/bobocode/bring/core/exception/PreDestroyException.java:4]: add public static final

File: core/src/main/java/com/bobocode/bring/core/exception/PreDestroyException.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.bobocode.bring.core.exception;
+
+public class PreDestroyException extends RuntimeException {
```
Comment (PolukovY) [inline core/src/main/java/com/bobocode/bring/core/exception/PreDestroyException.java:3]: Add JavaDoc

File: core/src/main/java/com/bobocode/bring/core/context/BringBeanFactory.java [hunk 4]
```diff
@@ -46,5 +46,7 @@ public interface BringBeanFactory {
      * @return A map containing all registered bean names as keys and corresponding bean instances as values.
      */
     <T> Map<String, T> getAllBeans();
+
+    void close();
```
Comment (PolukovY) [inline core/src/main/java/com/bobocode/bring/core/context/BringBeanFactory.java:50]: Add JavaDoc to method

File: core/src/main/java/com/bobocode/bring/core/annotation/PreDestroy.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.bobocode.bring.core.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface PreDestroy {
```
Comment (PolukovY) [inline core/src/main/java/com/bobocode/bring/core/annotation/PreDestroy.java:10]: Add JavaDoc to annotation

File: core/src/test/java/com/bobocode/bring/core/predestroy/PredestroyTest.java (hunk 1 above)
Comment (proyalko) [reply core/src/test/java/com/bobocode/bring/core/predestroy/PredestroyTest.java:50]: fixed

File: core/src/main/java/com/bobocode/bring/core/exception/PreDestroyException.java (hunk 3 above)
Comment (proyalko) [reply core/src/main/java/com/bobocode/bring/core/exception/PreDestroyException.java:3]: done

File: core/src/main/java/com/bobocode/bring/core/annotation/PreDestroy.java (hunk 5 above)
Comment (proyalko) [reply core/src/main/java/com/bobocode/bring/core/annotation/PreDestroy.java:10]: done

**Review by PolukovY — APPROVED** [review]

**Pull request merged**